kivy/properties.pyx

```diff
@@ -1996,34 +2003,38 @@ cdef class ColorProperty(Property):
         Color value will be dispatched when set through indexing or slicing,
         but when setting with slice you must ensure that slice has 4 components
         with float values between 0-1.
+        Assingning color name as value is now supported.
+        Value `None` is allowed as default value for property.
     '''
 
-    def __init__(self, defaultvalue=None, **kw):
-        defaultvalue = defaultvalue or [1, 1, 1, 1]
+    def __init__(self, defaultvalue=0, **kw):
+        defaultvalue = \
+            [1.0, 1.0, 1.0, 1.0] if defaultvalue == 0 else defaultvalue
         super(ColorProperty, self).__init__(defaultvalue, **kw)
 
     cdef convert(self, EventDispatcher obj, x):
         if x is None:
             return x
-        if isinstance(x, string_types):
-            return ObservableList(self, obj, self.parse_str(obj, x))
-
+        cdef object color = x
         try:
-            count = len(x)
-        except TypeError as e:
+            if isinstance(x, string_types):
+                color = self.parse_list(obj, self.parse_str(obj, x))
+            color = self.parse_list(obj, color)
```

Comment:
You're parsing list twice if it's a string.

Reply:
Right, for sting case it should be ```color = self.parse_str(obj, x)``` and `color` will be a `list` and `parse_list` then can return `ObservableList`.